### PR TITLE
#12809 - Add support for HIDE_JURISDICTION feature to campaigns

### LINKED ISSUE #12809: Add support for the hide jurisdiction fields feature to campaigns [1]
### Feature Description

The option to hide jurisdiction fields introduced in https://github.com/SORMAS-Foundation/SORMAS-Project/issues/12550 needs to be applied to the campaign module as well.

### Added Value/Benefit

Countries that don't use jurisdictions will be able to use the campaign module.

### Acceptance Criteria

- [ ] If the `HIDE_JURISDICTION` feature is enabled, hide all jurisdiction fields and automatically fill the region, district, and community fields with the respective default options by using the getDefaultInfrastructure method (see CaseCreateForm and CaseDataForm)
- [ ] If the feature is enabled, the jurisdiction filters and columns in the campaign directories are hidden

### Implementation Details

_No response_

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by carina29**
Fixes #12809

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12896

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12896

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12896

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12896

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12896

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12896

File: sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaigndata/CampaignDataGrid.java [hunk 1]
```diff
@@ -63,6 +64,12 @@ protected void addDefaultColumns() {
 		for (Column<?, ?> column : getColumns()) {
 			column.setCaption(I18nProperties.getPrefixCaption(CampaignFormDataIndexDto.I18N_PREFIX, column.getId(), column.getCaption()));
 		}
+
+		if (FacadeProvider.getFeatureConfigurationFacade().isFeatureEnabled(FeatureType.HIDE_JURISDICTION_FIELDS)) {
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaigndata/CampaignDataGrid.java:68]: Replace with UiUtil.enabled

File: sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaigndata/CampaignFormDataEditForm.java [hunk 2]
```diff
@@ -145,6 +150,23 @@ protected void addFields() {
...
+
+		cbDistrict.setVisible(false);
+		cbDistrict.setValue(FacadeProvider.getDistrictFacade().getDefaultInfrastructureReference());
+
+		cbCommunity.setVisible(false);
+		cbCommunity.setValue(FacadeProvider.getCommunityFacade().getDefaultInfrastructureReference());
+
+		if (FacadeProvider.getFeatureConfigurationFacade().isFeatureEnabled(FeatureType.INFRASTRUCTURE_TYPE_AREA)) {
+			cbArea.setVisible(false);
+			cbArea.setValue(FacadeProvider.getRegionFacade().getByUuid(defaultRegion.getUuid()).getArea());
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaigndata/CampaignFormDataEditForm.java:166]: Will throw a NPE if there is no default region; there will be an error when trying to submit the form anyway in this case, but I'd rather not have an exception thrown here already

File: sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaignstatistics/CampaignStatisticsGrid.java [hunk 3]
```diff
@@ -49,7 +50,9 @@ public void setDataProvider() {
 	}
 
 	public void setColumnsVisibility(CampaignJurisdictionLevel groupingLevel) {
-		setAreaColumnVisible(CampaignJurisdictionLevel.AREA.equals(groupingLevel) || CampaignJurisdictionLevel.REGION.equals(groupingLevel));
+		setAreaColumnVisible(
+			!FacadeProvider.getFeatureConfigurationFacade().isFeatureEnabled(FeatureType.HIDE_JURISDICTION_FIELDS)
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaignstatistics/CampaignStatisticsGrid.java:54]: Use UiUtil.disabled

File: sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaignstatistics/CampaignStatisticsView.java [hunk 4]
```diff
@@ -80,18 +81,21 @@ public CampaignStatisticsView() {
...
-		jurisdictionSelector.addValueChangeListener(e -> {
-			CampaignJurisdictionLevel groupingValue = (CampaignJurisdictionLevel) e.getValue();
-			criteria.setGroupingLevel(groupingValue);
-			grid.setColumnsVisibility(groupingValue);
-			grid.reload();
-		});
-		jurisdictionLayout.addComponent(jurisdictionSelector);
-		mainLayout.addComponent(jurisdictionLayout);
 
+		if (FacadeProvider.getFeatureConfigurationFacade().isFeatureEnabled(FeatureType.HIDE_JURISDICTION_FIELDS)) {
```
Comment (MateStrysewske) [inline sormas-ui/src/main/java/de/symeda/sormas/ui/campaign/campaignstatistics/CampaignStatisticsView.java:86]: Use UiUtil.enabled

**Review by MateStrysewske — CHANGES_REQUESTED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12896

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12896

**Review by MateStrysewske — APPROVED** [review]

**Pull request merged**